### PR TITLE
👽️ Smart Contract - Upgrade cairo lang to latest version

### LINKED ISSUE #55: Smart Contract - Upgrade cairo-lang to latest version
**Description**
---
A 0.10 version has been released, we need to upgrade our contracts accordingly.

**What code could be improved?**
---
All cairo files.

**Acceptance Criteria**
---

- [ ] Tests works with latest Cairo version.

**Additional context**
---
- Release note with script to migrate: https://github.com/starkware-libs/cairo-lang/releases/tag/v0.10.0
- Prerequisites: OZ migration, Protostar migration.

### PR DESCRIPTION
**Pull request by Bal7hazar**
Fix #55 

- [x] Cairo from 0.9.1 to 0.10.0
- [x] protostar from 0.3.3 to 0.4.2
- [x] OZ cairo contracts from 0.3.2 to 0.4.0b
- [x] CairOpen (latest main)
- [x] OZ fork (latest branch)

### CODE REVIEW

**Comment by Bal7hazar** [comment]
This PR introduce a regression due to protostar v0.4.2 since it does not use latest starknet version, the next 0.4.3 version must do.
The features impacted are starknet cli usage interacting with smart contracts (such as invoke used for transferOwnership) in our scripts.

We decided to stay like this until the next protostar version.

File: src/badge/badge.cairo [hunk 1]
```diff
@@ -1,218 +1,218 @@
-# SPDX-License-Identifier: MIT
-# Carbonable Contracts written in Cairo v0.9.1 (badge.cairo)
+// SPDX-License-Identifier: MIT
+// Carbonable Contracts written in Cairo v0.9.1 (badge.cairo)
```
Comment (cloudvenger) [inline src/badge/badge.cairo:2]: The reference should be Cairo v0.10 (badge.cairo). no ?

File: src/badge/library.cairo [hunk 2]
```diff
@@ -1,172 +1,172 @@
-# SPDX-License-Identifier: MIT
-# OpenZeppelin Contracts for Cairo v0.2.1 (token/erc1155/library.cairo)
+// SPDX-License-Identifier: MIT
+// OpenZeppelin Contracts for Cairo v0.2.1 (token/erc1155/library.cairo)
```
Comment (cloudvenger) [inline src/badge/library.cairo:2]: Change the reference from OpenZeppelin to the repo Bal7hazar. No ?

File: src/interfaces/badge.cairo [hunk 3]
```diff
@@ -1,45 +1,45 @@
-# SPDX-License-Identifier: MIT
-# Carbonable Contracts written in Cairo v0.9.1 (project.cairo)
+// SPDX-License-Identifier: MIT
+// Carbonable Contracts written in Cairo v0.9.1 (project.cairo)
```
Comment (cloudvenger) [inline src/interfaces/badge.cairo:2]: Same here: The reference should be Cairo v0.10 (badge.cairo). no ?

File: src/interfaces/project.cairo [hunk 4]
```diff
@@ -1,133 +1,133 @@
-# SPDX-License-Identifier: MIT
-# Carbonable Contracts written in Cairo v0.9.1 (project.cairo)
+// SPDX-License-Identifier: MIT
+// Carbonable Contracts written in Cairo v0.9.1 (project.cairo)
```
Comment (cloudvenger) [inline src/interfaces/project.cairo:2]: Same here: The reference should be Cairo v0.10 (badge.cairo). no ?

**Comment by cloudvenger** [comment]
I did not comment each line, but generally in all code, do we need to upgrade the reference of the version of Cairo in the code ?
Change :
`// Carbonable Contracts written in Cairo v0.9.1 (library.cairo)`
into 
`// Carbonable Contracts written in Cairo v0.10 (library.cairo)`

**Comment by Bal7hazar** [comment]
> I did not comment each line, but generally in all code, do we need to upgrade the reference of the version of Cairo in the code ? Change : `// Carbonable Contracts written in Cairo v0.9.1 (library.cairo)` into `// Carbonable Contracts written in Cairo v0.10 (library.cairo)`

Yes I'll, actually I'm not sure if it is the Cairo version or our contract repo version since it is not clear from OZ repo imo.
Maybe we could just remove this useless comment? (the whole line).

**Review by cloudvenger — APPROVED** [review]

**Pull request merged**